libs/store/src/lib/effects/race.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  AppPlayerId,
+  AppPlayerProfiles,
+  AppRace,
+  AppRaceId,
+  AppTournament,
+  AppTournamentId,
+  AppTournamentState,
+} from '@razor/models';
+import {
+  calculateTimeoutTimer,
+  extractId,
+  extractIdType,
+  generateLeaderboard,
+  giveZeroPadding,
+} from '@razor/util';
+import { playerNotFound, tournamentNotFound } from '../loggers';
+import { raceNotFound } from '../loggers/race';
+import {
+  endCountdownPayload,
+  endRacePayload,
+  startCountdownPayload,
+} from '../payloads';
+import { Dispatch, RootState } from '../store';
+
+//TODO: should move this to server later
+const loadRacingText = async (): Promise<string> => {
+  const url = 'http://www.metaphorpsum.com/paragraphs/1/8';
+
+  return fetch(url)
+    .then(response => response.text())
+    .then(data => {
+      return data;
+    });
+};
+
+export const startCountdown = async (
+  dispatch: Dispatch,
+  payload: startCountdownPayload,
+  state: RootState,
+): Promise<void> => {
+  const {
+    tournamentId,
+    playerId,
+  }: { tournamentId: AppTournamentId; playerId: AppPlayerId } = payload;
+
+  if (!state.game.tournamentsModel[tournamentId])
+    tournamentNotFound(dispatch, tournamentId, `Started by: ${playerId}`);
+  if (!state.game.playersModel[playerId])
+    playerNotFound(
+      dispatch,
+      playerId,
+      `While tournament starting: ${tournamentId}`,
+    );
+
+  const numberOfRacesBefore =
+    state.game.tournamentsModel[tournamentId] &&
+    state.game.tournamentsModel[tournamentId].raceIds.length;
+  const raceIndex = numberOfRacesBefore
+    ? giveZeroPadding(numberOfRacesBefore.toString(), 3)
+    : '000';
```

Comment:
what's this one doing?

Reply:
Fix in PR #14 under commit 027ab37b45b045ae9328637779ce152f7989e5dd